File: Intro_to_C_and_CPlusPlus/CPlusPlus/lab/lab9/python/Date.py
# Basic class definition
class Date(object):
    DATES = ["Jan", "Feb", "Mar", "Apr",
              "May", "Jun", "Jul", "Aug",
              "Sep", "Oct", "Nov", "Dec"]

    # Constructor
    def __init__(self, month, day, year):
        self.month = month
        self.day = day
        self.year = year

    # toString
    def to_s(self):
        return "{0} {1}, {2}".format(Date.DATES[self.month-1], self.day, self.year)

    # ================== Fix Me! ================
    # Am I older than Date d?
    def older(self, d):
        # TODO: Implement
    # ================== ^^^^^^ ================


# Make some Dates
d1 = Date(10,5,1993)
d2 = Date(8,3,1895)
d3 = Date(4,3,2000)

# Make them into an array
dates = [d1,d2,d3]
oldest_date = dates[0]

# Loop over the dates and find the oldest
for date in dates:
    if date.older(oldest_date):
        oldest_date = date

print oldest_date.to_s()

File: Intro_to_C_and_CPlusPlus/CPlusPlus/lab/lab9/ruby/Date.rb
#  Basic Class definition in Ruby
class Date
  # This makes getters and setters for us!
  attr_accessor :month, :day, :year
  @@months = ["Jan", "Feb", "Mar", "Apr",
              "May", "Jun", "Jul", "Aug",
              "Sep", "Oct", "Nov", "Dec"]

  # Constructor
  def initialize(month, day, year)
    @month = month
    @day = day
    @year = year
  end

  # This is a toString method!
  # In Ruby, the last statement executed in a method is the return value if
  # there is no explicit return
  def to_s
    "#{@@months[@month-1]} #{@day}, #{@year}"
  end

  # ================== Fix Me! ================
  # Is the given date older than me?
  def older?(d)
    # TODO: Implement me!
  end
  # ================== ^^^^^^ ================

end

# Make some Dates
d1 = Date.new(10,5,1993)
d2 = Date.new(8,3,1895)
d3 = Date.new(4,3,2000)

# Make them into an array
dates = [d1,d2,d3]
oldest_date = dates[0]

# Loop over the dates and find the oldest
for date in dates do
  oldest_date = date if date.older?(oldest_date)
end

puts oldest_date
